homeassistant/components/vulcan/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Vulcan component."""
+import logging
+
+from aiohttp import ClientConnectorError
+from vulcan import Account, Keystore, Vulcan
+from vulcan._utils import VulcanAPIException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["calendar"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Uonet+ Vulcan integration."""
+    hass.data.setdefault(DOMAIN, {})
+    try:
+        keystore = Keystore.load(entry.data["keystore"])
+        account = Account.load(entry.data["account"])
+        client = Vulcan(keystore, account)
```

Comment:
If the client is using aiohttp we prefer that libraries accept the Home Assistant aiohttp client session as an optional parameter to use for the requests.

https://github.com/home-assistant/core/blob/92f1855bcfaad514aaa9614a91a426c2e52402ce/homeassistant/helpers/aiohttp_client.py#L37-L42